site/src/main/java/uk/nhs/digital/ps/components/SearchComponent.java

```diff
@@ -0,0 +1,79 @@
+package uk.nhs.digital.ps.components;
+
+import org.apache.commons.lang.StringUtils;
+import org.hippoecm.hst.component.support.bean.BaseHstComponent;
+import org.hippoecm.hst.content.beans.query.HstQuery;
+import org.hippoecm.hst.content.beans.query.HstQueryResult;
+import org.hippoecm.hst.content.beans.query.exceptions.QueryException;
+import org.hippoecm.hst.content.beans.query.filter.Filter;
+import org.hippoecm.hst.content.beans.standard.HippoBean;
+import org.hippoecm.hst.core.component.HstComponentException;
+import org.hippoecm.hst.core.component.HstRequest;
+import org.hippoecm.hst.core.component.HstResponse;
+import org.hippoecm.hst.core.parameters.ParametersInfo;
+import org.hippoecm.hst.core.request.HstRequestContext;
+import org.hippoecm.hst.util.SearchInputParsingUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.nhs.digital.ps.beans.Publication;
+import java.util.Calendar;
+
+@ParametersInfo(type = SearchComponentInfo.class)
+public class SearchComponent extends BaseHstComponent {
+
+    public static final Logger Log = LoggerFactory.getLogger(SearchComponent.class);
+
+    @Override
+    public void doBeforeRender(final HstRequest request, final HstResponse response) throws HstComponentException {
+
+        HstRequestContext requestContext = request.getRequestContext();
+        HippoBean scopeBean = requestContext.getContentBean();
+
+        SearchComponentInfo info = getComponentParametersInfo(request);
+
+        try {
+
+            HstQuery query = requestContext.getQueryManager().createQuery(scopeBean, Publication.class);
+            query.setLimit(info.getLimit());
+
+            int offset = 0;
+
+            try {
+                String offsetString = request.getParameter("offset");
+                offset = Integer.parseInt(offsetString);
+
+            } catch (NumberFormatException e) {
+                // No offset set in this request, so use default 0
+            }
```

Comment:
How about extracting this logic into a private method and call it here as `int offset = extractOffsetFrom(request);`? 

Maybe it's just me, but, given that this whole try/catch block takes quite a few lines, it breaks the flow when reading this method.

Reply:
done in my branch for Search :)

Reply:
Looks like this block will change going forward.  With the Pagination ticket I'm looking at, the offset is set in a different way.